alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/navigationstrategies/Exploring.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.actions.navigationstrategies
+
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
+import it.unibo.alchemist.model.interfaces.EuclideanNavigationAction
+import it.unibo.alchemist.model.interfaces.EuclideanNavigationStrategy
+import it.unibo.alchemist.model.interfaces.NavigationStrategy
+import it.unibo.alchemist.model.interfaces.OrientingPedestrian
+import it.unibo.alchemist.model.interfaces.Pedestrian
+import it.unibo.alchemist.model.interfaces.geometry.euclidean.twod.ConvexPolygon
+import it.unibo.alchemist.model.interfaces.geometry.euclidean.twod.Euclidean2DConvexShape
+import it.unibo.alchemist.model.interfaces.geometry.euclidean.twod.Euclidean2DTransformation
+import it.unibo.alchemist.model.interfaces.graph.Euclidean2DPassage
+import java.awt.Shape
+import kotlin.math.abs
+import kotlin.math.pow
+
+/**
+ * A [NavigationStrategy] allowing to explore the environment.
+ * In order to choose which direction to take, a weighting system is used: every time the
+ * pedestrian enters a new room all the visible doors are weighted, the one with minimum
+ * weight is then crossed. The weighting system used here is derived from the one by
+ * [Andresen et al.](https://doi.org/10.1080/23249935.2018.1432717), see [weight].
+ *
+ * @param T the concentration type.
+ * @param N the type of landmarks of the pedestrian's cognitive map.
+ * @param E the type of edges of the pedestrian's cognitive map.
+ */
+open class Exploring<T, N : Euclidean2DConvexShape, E>(
+    override val action: EuclideanNavigationAction<T, N, E, ConvexPolygon, Euclidean2DPassage>,
+    /**
+     * Weight assigned to known impasses, see [impasseFactor].
+     */
+    private val impasseWeight: Double = DEFAULT_IMPASSE_WEIGHT
+) : EuclideanNavigationStrategy<T, N, E, ConvexPolygon, Euclidean2DPassage> {
+
+    companion object {
+        /**
+         * Empirically found to work well (allows the pedestrian to avoid known impasses).
```

Comment:
Please add a note on the wiki as well.
Maybe a tabular with all the hardcoded parameters and their meaning? I do not want them to slip away when we write the paper

Reply:
I added two tables: one describing the factors used by the weighting system, another containing all the hardcoded parameters.